### PR TITLE
feat(design): add default theme and update docs

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
There is no default theme set in `@daffodil/design`.

Fixes: #2172 
Part of: #336 

## What is the new behavior?
Set a predefined theme in `@daffodil/design` so that users do not have to configure a theme if they choose to use Daffodil's default theme.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@xelaint there's a ref in the design `getting-started` to the typography. I'm not sure the getting started still makes sense as written.

File: libs/design/scss/theme.scss [hunk 1]
```diff
@@ -9,12 +9,11 @@
 // compiled stylesheet size.
 
 // @usage
-// ```
+// ```scss
 // @use '@daffodil/design/scss/theme';
 // ```
 
 @forward 'core';
-@forward 'accessibility';
```
Comment (damienwebdev) [inline libs/design/scss/theme.scss:17]: Why was this removed?

File: libs/design/scss/theming/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Theming
+# Daffodil Theming
 Daffodil's theming capabilities enables you to customize `@daffodil/design` components to reflect your brand. A theme consists of custom color configurations that will work in dark and light themes.
 
 ## Custom Colors
```
Comment (damienwebdev) [inline libs/design/scss/theming/README.md:4]: This is broken.

Comment (damienwebdev) [inline libs/design/scss/theming/README.md:4]: There's a link here for `[validate-palette-function]` that goes nowhere, it should be removed.

File: libs/design/scss/theming/README.md [hunk 3]
```diff
@@ -9,23 +9,48 @@ Daffodil's theming capabilities enables you to customize `@daffodil/design` comp
 ## Themes
 Dark and light modes are supported in all `@daffodil/design` components. When a theme is not specified, Daffodil defaults to the `light` mode.
 
-### Setting up the theme file
-Configure your application to support light and dark modes by adding the following to the `theme.scss` file:
+## Palettes
+A palette is a collection of perceptually uniform colors with consistent contrast ratios. `@daffodil/design`'s color palettes are represented by a Sass map, with each value in a palette called a **hue**.
```
Comment (damienwebdev) [inline libs/design/scss/theming/README.md:13]: Define perceptually uniform or add a link?

File: libs/design/scss/theming/README.md [hunk 4]
```diff
@@ -34,20 +59,42 @@ $white: daff-theme.daff-map-deep-get($theme, 'core.white');
...
-### Setting up the styles file
-Add `theme.scss` to the `styles.scss` file.
+### Setting up the styles file with `@daffodil/design`'s theme
+Use `@daffodil/design`'s theme module to the `styles.scss` file:
+
+```scss
+@use '@daffodil/design/scss/theme' as daff-theme;
+```
+
+Create classes in the `styles.scss` file to include the `theme` module for `$theme` and `$theme-dark` variables. This will allow you to set a click event on a button to switch between modes. [View this setup in Stackblitz](https://stackblitz.com/edit/daffodil-design-theming-angular-10)
```
Comment (damienwebdev) [inline libs/design/scss/theming/README.md:69]: The `stackblitz` link looks stale.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

**Comment by xelaint** [comment]
@damienwebdev yeah, that section doesn't really make sense anymore. I can remove it.

File: libs/design/scss/theming/README.md (hunk 2 above)
Comment (xelaint) [reply libs/design/scss/theming/README.md:4]: What do you mean broken?

Comment (damienwebdev) [reply libs/design/scss/theming/README.md:4]: The link. GH didn't do a good job referencing the line:

:stop: Before you begin, read the [accessibility guide on color in @daffodil/design](https://github.com/graycoreio/daffodil/blob/43048f9ee6bc0753672aa8f8bb7674f8878a062e/libs/design/guides/color#accessibility.md).

File: libs/design/scss/theming/README.md (hunk 3 above)
Comment (xelaint) [reply libs/design/scss/theming/README.md]: updated

**Review by damienwebdev — APPROVED** [review]
